src/System.IO.FileSystem.DriveInfo/src/System/IO/DriveInfo.Windows.cs

```diff
@@ -12,8 +13,7 @@ public sealed partial class DriveInfo
     {
         private static string NormalizeDriveName(string driveName)
         {
-            if (driveName == null)
-                throw new ArgumentNullException("driveName");
+            Debug.Assert(driveName != null);
```

Comment:
You can get rid of the Contract.EndContractBlock() as well.  Even if we were using contracts, EndContractBlock only applies after certain if/throw patterns.